src/main/java/duke/Ui.java

```diff
@@ -0,0 +1,45 @@
+package duke;
+
+import java.util.Scanner;
+
+/**
+ * Represents the user interface for interacting with the chatbot.
+ * This class handles input and output operations, providing methods for
+ * reading user commands, displaying messages, and presenting a consistent
+ * interface to the user.
+ */
+public class Ui {
+    private Scanner scanner;
+    public Ui() {
+        scanner = new Scanner(System.in);
+    }
+
+    public void breakLine() {
+        System.out.println("---------------------------------------");
+    }
+
+    /**
+     * Prints a greeting to the user.
+     */
+    public void greet() {
```

Comment:
Small detail but since the method mainly displays/prints  like the `showMessage()` method, perhaps can consider grouping these methods together and give them similar naming, e.g. `showGreetings()` or `showWelcomeMessage()` !